### PR TITLE
Add data-testids

### PR DESCRIPTION
**Pull request by aacic**
## Description

Added data-testids to html and svg tags to support e2e tests.

Changes to container/ci/serverconfig.json should be ignore, I'll revert them before we merge this branch.

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: client/mds3/sampletable.js [hunk 1]
```diff
@@ -161,6 +163,7 @@ async function feedSample2selectCallback(tk, block, _samples, sampleIdxLst) {
 
 async function make_singleSampleTable(s, arg) {
 	const table = arg.singleSampleDiv || table2col({ holder: arg.div })
+	table.attr()
```
Comment (xzhou82) [inline client/mds3/sampletable.js:166]: purpose?

File: client/plots/plot.ssgq.js [hunk 2]
```diff
@@ -65,7 +65,10 @@ export async function plotSingleSampleGenomeQuantification(
 		const q2 = termdbConfig.queries.singleSampleGbtk?.[q.singleSampleGbtk]
 
 		// description
-		holder.append('div').text(q.description || queryKey)
+		holder
+			.append('div')
+			.attr('data-testid', 'methylation_array_sand_box')
```
Comment (xzhou82) [inline client/plots/plot.ssgq.js:70]: methylation array is just one type of data that could be shown in ssgq. i prefer to rename id to `sjpp_ssgq_sandbox`

File: client/src/block.js [hunk 3]
```diff
@@ -283,9 +283,12 @@ export class Block {
 				.append('div') // contains svg and resize button
 				.style('position', 'relative')
 				.style('display', 'inline-block')
-			this.svg = inlinediv.append('svg').on('mousedown', event => {
-				event.preventDefault()
-			})
+			this.svg = inlinediv
+				.append('svg')
+				.attr('data-testid', 'sja_plot_svg')
```
Comment (xzhou82) [inline client/src/block.js:288]: rename to sjpp_block_svg

File: client/mds3/sampletable.js [hunk 4]
```diff
@@ -73,7 +73,7 @@ export async function displaySampleTable(samples, args) {
 		return await make_singleSampleTable(samples[0], args)
 	}
 	const [columns, rows] = await samples2columnsRows(samples, args.tk)
-	const params = { rows, columns, div: args.div, resize: rows.length > 10 }
+	const params = { rows, columns, div: args.div, resize: rows.length > 10, dataTestId: 'sja_simpletable' }
```
Comment (xzhou82) [inline client/mds3/sampletable.js:76]: rename id to sjpp_mds3tk_sampletable

File: client/mds3/leftlabel.variant.js [hunk 5]
```diff
@@ -165,6 +169,7 @@ function menu_variants(tk, block) {
 			.append('div')
 			.text('Download')
 			.attr('class', 'sja_menuoption')
+			.attr('data-testid', 'sja_download_menuoption')
```
Comment (xzhou82) [inline client/mds3/leftlabel.variant.js:172]: sjpp_mds3tk_variantdownload_menuoption

File: client/mds3/leftlabel.sample.js [hunk 6]
```diff
@@ -29,6 +29,7 @@ export function makeSampleLabel(data, tk, block, laby) {
 			.attr('class', 'sja_clbtext2')
 			.style('opacity', 1)
 			.text(`${data.sampleTotalNumber} sample${data.sampleTotalNumber > 1 ? 's' : ''}`)
+			.attr('data-testid', 'sja-samples-label')
```
Comment (xzhou82) [inline client/mds3/leftlabel.sample.js:32]: please let me know if it's worth to "scope" the ids by app/component name prefix, e.g. all ids assigned to scripts in client/mds3/* are prefixed by `sjpp-mds3tk-`

File: client/mds3/sampletable.js (hunk 4 above)
Comment (aacic) [reply client/mds3/sampletable.js]: Change

File: client/plots/plot.ssgq.js (hunk 2 above)
Comment (aacic) [reply client/plots/plot.ssgq.js]: Changed.

File: client/mds3/leftlabel.variant.js (hunk 5 above)
Comment (aacic) [reply client/mds3/leftlabel.variant.js]: Changed.

File: client/mds3/leftlabel.sample.js (hunk 6 above)
Comment (aacic) [reply client/mds3/leftlabel.sample.js]: I've changed the label: `sja-samples-label` to `sjpp_mds3tk_samples_label`.
We need a clear standard on how to name `data-testids`. On the other hand, this is still a "proof of concept", and if it turns out that the approach is right and we continue to add e2e tests, then we can rename existing `data-testids`. There are just about 15 of them and it's easy to find them in the code. 
I would suggest we test this in prod first, to see how it works.

Comment (xzhou82) [reply client/mds3/leftlabel.sample.js]: sure thanks. also i prefer dash over underscore

**Review by xzhou82 — APPROVED** [review]

**Pull request merged**